### ISSUE TITLE
ELC with dielectric contrasts doesn't work

### ISSUE THREAD

**Issue by fweik** [espressomd/espresso#229]
Mon 04 Aug 2014 03:44:21 PM CEST, original submission:
Errors are not huge, but very noticable with single test charges. I fixed one obvious bug which at least in some cases fixes the forces, but so far, don't trust ELC with any dielectrics.

**Comment by RudolfWeeber** [comment]
@KonradBreitsprecher: Is this still broken?

**Comment by arnolda** [comment]
Well, there are no further relevant commits on ELC after the Aug 2014 fixes, so I would say yes.

**Comment by KonradBreitsprecher** [comment]
So I digged into this again, the following script sometimes fails and sometimes works, even though p3m tunes to the same values. Note that in the current branch, the dielectric contrast feature is not interfaced.

```

import numpy as np
import espressomd
from espressomd.electrostatics import P3M, P3M_GPU
from espressomd import electrostatic_extensions

S = espressomd.System()
box_l = 10.0
bl2 = box_l/2.0
elc_gap = 5.0
box_z = box_l+elc_gap
S.box_l = [box_l, box_l, box_z]
S.time_step = 0.001
S.cell_system.skin=0.4

d=0.8
S.part.add(pos = [bl2, bl2, box_l-d], q = 1, fix = [1,1,1])
S.part.add(pos = [bl2, bl2/2.0, bl2], q = -0.5, fix=[1,1,1])
S.part.add(pos = [bl2, bl2*3.0/2.0, bl2], q = -0.5, fix=[1,1,1])
p3m = P3M(bjerrum_length=1.0, accuracy=1e-3)
elc = electrostatic_extensions.ELC(gap_size = elc_gap, maxPWerror = 1e-4, di_mid_top = 1000, di_mid_bot = 1000)
S.actors.add(p3m)
S.actors.add(elc)

n=100
ret = []
for i in range(n):
    z= box_l-d - 1.0*i/n*(box_l-2*d)
    S.part[0].pos = [bl2, bl2, z]
    S.integrator.run(0)
    ret.append([z, S.part[0].f[2]])
np.savetxt("elc.dat", ret)
                                             
```                                                                              
Wrong:
![image](https://user-images.githubusercontent.com/3974659/31783004-09a5d56c-b4fd-11e7-9c9f-468364e12e81.png)
Correct:
![image](https://user-images.githubusercontent.com/3974659/31783085-514cd5c8-b4fd-11e7-94b1-11a39b3fee8e.png)

**Comment by RudolfWeeber** [comment]
@KonradBreitsprecher is this resolved?

**Comment by KonradBreitsprecher** [comment]
Yes! Is tested and can be closed via #1638
